librz/analysis/esil/esil_trace.c

```diff
@@ -175,11 +199,21 @@ static int trace_hook_mem_write(RzAnalysisEsil *esil, ut64 addr, const ut8 *buf,
 	size_t i;
 	int ret = 0;
 	char *hexbuf = malloc((1 + len) * 3);
-	sdb_array_add_num(DB, KEY("mem.write"), addr, 0);
+
+	// convert data to ut64
+	// FIXME : a better way to convert between them or change the argument
 	rz_hex_bin2str(buf, len, hexbuf);
-	sdb_set(DB, KEYAT("mem.write.data", addr), hexbuf, 0);
-	//eprintf ("[ESIL] MEM WRITE 0x%08"PFMT64x" %s\n", addr, hexbuf);
+	ut64 val = strtol(hexbuf, NULL, 16);
```

Comment:
see the comment about the `value` field of RzILTraceMemOp. It should probably be possible to just copy the buf data into RzILTraceMemOp->value.